src/main/java/com/cloudbees/plugins/credentials/impl/BaseStandardCredentials.java

```diff
@@ -161,7 +175,28 @@ private static FormValidation checkForDuplicates(String value, ModelObject conte
             return null;
         }
 
-        public final FormValidation doCheckId(@QueryParameter String value, @AncestorInPath ModelObject context) {
+        /**
+         * Gets the check id url for the specified store.
+         *
+         * @param store the store.
+         * @return the url of the id check endpoint.
+         * @throws UnsupportedEncodingException if the JVM does not implement the JLS.
+         */
+        public String getCheckIdUrl(CredentialsStore store) throws UnsupportedEncodingException {
+            ModelObject context = store.getContext();
+            for (ContextResolver r : ExtensionList.lookup(ContextResolver.class)) {
+                String token = r.getToken(context);
+                if (token != null) {
+                    return Jenkins.getActiveInstance().getRootUrlFromRequest() + "/descriptorByName/" + clazz.getName()
+                            + "/checkId?provider=" + r.getClass().getName() + "&token="
```

Comment:
talk to core! (but I'll switch to sharing core's bug so I can share in core's fix)